airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/demo/file_based/platform/PlatformStuff.java

```diff
@@ -0,0 +1,103 @@
+package io.airbyte.integrations.destination.snowflake.demo.file_based.platform;
+
+import static java.util.stream.Collectors.toMap;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.integrations.base.destination.typing_deduping.CatalogParser;
+import io.airbyte.integrations.base.destination.typing_deduping.ParsedCatalog;
+import io.airbyte.integrations.base.destination.typing_deduping.StreamConfig;
+import io.airbyte.integrations.base.destination.typing_deduping.StreamId;
+import io.airbyte.integrations.destination.snowflake.demo.file_based.iface.StreamDestination;
+import io.airbyte.integrations.destination.snowflake.demo.file_based.iface.StreamDestinationFactory;
+import io.airbyte.protocol.models.v0.AirbyteMessage;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class PlatformStuff {
+
+  private final ParsedCatalog catalog;
+  private final JsonNode config;
+  private final StreamDestinationFactory streamDestinationFactory;
+  private final Supplier<RecordWriter> recordWriterFactory;
+  private final int targetByteSizePerFile;
+
+  // There are probably a ton more parameters here. In reality we would make this a builder.
```

Comment:
for example, destinations might want to pass in a `defaultNamespace` parameter. Then platform could guarantee that destinations never sees a `namespace=null` record message, and always construct StreamDestination objects using non-null namespaces.